Sources/GDBRemote/Mixins/FileOperationsMixin.hpp

```diff
@@ -130,18 +130,26 @@ ErrorCode FileOperationsMixin<T>::onQueryModuleInfo(Session &session,
                                                      std::string &triple,
                                                      ModuleInfo &info) const {
   ByteVector buildId;
-  if (Platform::GetExecutableFileBuildID(path, buildId)) {
-    // format the uuid as an upper-case string with two hex chars per byte
-    std::ostringstream ss;
-    for(const auto b : buildId)
-      ss << std::uppercase << std::hex << std::setfill('0') << std::setw(2) << int(b);
-
-    info.uuid = ss.str();
+  if (!Platform::GetExecutableFileBuildID(path, buildId)) {
+    // Not all executable files contain an embedded build ID. In this case,
+    // return a crc32 of the contents of the file. The documentation for
+    // `qModuleInfo` suggests an md5 hash can be returned as "md5" in the
+    // response instead of "uuid." However, returning a crc32 is consistent
+    // with behavior of lldb-server.
```

Comment:
Technically, MD5 is the right hash here. The default build ID hash was MD5, though it is not required to be MD5. The modern default has changed to SHA1, though `UUID` is becoming somewhat common too. The user also has the ability to specify a custom (hex) string of an even length as well. Is there some expectation that it is the CRC? I'd prefer MD5/SHA1 over CRC32.

Reply:
I would prefer to follow the docs and use md5 here, but there are a number of lldb tests that that start failing when we report an md5 hash instead of uuid. I don't know if sha1 is allowed-- there is nothing in the documentation.

This crc implementation is what I can observe lldb-server doing while running tests, and it gets the most tests passing. I will go back to md5 and see if there's anything more I can do to get tests passing.